### DISCUSSION TITLE
Add `outline-current`

### DISCUSSION THREAD

**Discussion by Larsklopstra** [tailwindlabs/tailwindcss#5037] [Ideas]
It would be nice to have `outline-current` in the base.

## Use cases

It would allow for accessible focus states whilst keeping flexibility. When inverting colors you'd have to switch `outline-white` to `outline-blank`. In most cases, the foreground color offers sufficient contrast, with `outline-current` you'd always have the same outline contrast as the foreground text.

### "Hard coded"

Accessible

```html
<div class="bg-blue-900 text-white">
  <a href="#" class="focus-visible:outline-white">Example 1</a>
<div>
```

Not accessible

```html
<div class="bg-blue-200 text-blue-900">
  <a href="#" class="focus-visible:outline-white">Example 1</a>
<div>
```

Ouch. We'd have to refactor `outline-white` to `outline-black` to make it accessible.

### Current color

Accessible

```html
<div class="bg-blue-900 text-white">
  <a href="#" class="focus-visible:outline-current">Example 1</a>
<div>
```

Also accessible

```html
<div class="bg-blue-200 text-blue-900">
  <a href="#" class="focus-visible:outline-current">Example 1</a>
<div>
```

## You can configure it?

Of course you can. But, since Tailwind CSS already offers `-current` for other classes it would be nice to also have it for `outline`.

**Comment by olets** [comment]
You can do this with the more flexible `ring` utilities, which support colors, opacity, offset, and width.

https://play.tailwindcss.com/gU6HStMY5a

Simplest usage:

```html
<div class="ring-1 ring-current">…</div>
```

https://tailwindcss.com/docs/ring-color

Of course this only applies if you don't _need_ to use `outline`.

**Reply by Larsklopstra** [reply]
Doesn't yield the same results since `outline-black` and `outline-white` are dotted
